azurerm/internal/services/network/resource_arm_virtual_hub_connection.go

```diff
@@ -0,0 +1,301 @@
+		if connection, _ := findVirtualHubConnection(name, connections); connection != nil {
+			return tf.ImportAsExistsError("azurerm_virtual_hub_connection", *connection.ID)
+		}
+	}
+
+	connection := network.HubVirtualNetworkConnection{
+		Name: utils.String(name),
+		HubVirtualNetworkConnectionProperties: &network.HubVirtualNetworkConnectionProperties{
+			RemoteVirtualNetwork: &network.SubResource{
+				ID: utils.String(d.Get("remote_virtual_network_id").(string)),
+			},
+			AllowHubToRemoteVnetTransit:         utils.Bool(d.Get("hub_to_vitual_network_traffic_allowed").(bool)),
+			AllowRemoteVnetToUseHubVnetGateways: utils.Bool(d.Get("vitual_network_to_hub_gateways_traffic_allowed").(bool)),
+			EnableInternetSecurity:              utils.Bool(d.Get("internet_security_enabled").(bool)),
+		},
+	}
+	connections = append(connections, connection)
+	virtualHub.VirtualHubProperties.VirtualNetworkConnections = &connections
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, id.Name, virtualHub)
+	if err != nil {
+		return fmt.Errorf("Error adding Connection %q to Virtual Hub %q (Resource Group %q): %+v", name, id.Name, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for addition of Connection %q to Virtual Hub %q (Resource Group %q): %+v", name, id.Name, resourceGroup, err)
+	}
+
+	virtualHub, err = client.Get(ctx, resourceGroup, id.Name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", id.Name, resourceGroup, err)
+	}
+	if virtualHub.VirtualHubProperties == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties` was nil", id.Name, resourceGroup)
+	}
+	if virtualHub.VirtualHubProperties.VirtualNetworkConnections == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties.VirtualNetworkConnections` was nil", id.Name, resourceGroup)
+	}
+
+	connections = *virtualHub.VirtualHubProperties.VirtualNetworkConnections
+	newConnection, _ := findVirtualHubConnection(name, connections)
+	if newConnection == nil {
+		return fmt.Errorf("Connection %q was not found in Virtual Hub %q / Resource Group %q", name, id.Name, resourceGroup)
+	}
+	if newConnection.ID == nil && *newConnection.ID == "" {
+		return fmt.Errorf("Error retrieving Connection %q (Virtual Hub %q / Resource Group %q): `id` was nil or empty", name, id.Name, resourceGroup)
+	}
+
+	d.SetId(*newConnection.ID)
+	return resourceArmVirtualHubConnectionRead(d, meta)
+}
+
+func resourceArmVirtualHubConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.VirtualHubClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ParseVirtualHubConnectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	virtualHubName := id.VirtualHubName
```

Comment:
i don't think we need to assign these to locals?

Reply:
updated